### DISCUSSION TITLE
Friend-or-Foe disposition

### DISCUSSION THREAD

**Discussion by kostmo** [swarm-game/swarm#1539] [Ideas]
As mentioned in https://github.com/swarm-game/swarm/issues/1535#issuecomment-1722756888, in may be an interesting mechanic to restrict commands that act on other robots based on that robot's disposition towards us.  Perhaps there may be three possible dispositions:
* `Friend`
* `Neutral`
* `Foe`

In particular, the `give` command should probably be restricted if the target is a `Foe`.  In [`"Make a Friend"`](https://github.com/swarm-game/swarm/blob/ab7e5b6d8b79f23057ee49a65af0c2cf7d49e5ec/data/scenarios/Challenges/friend.yaml), on the other hand, the cat would initially be `Neutral` and receptive to `give`, and perhaps even become `Friend` after the gift.

Basically, a `Foe` would not permit you to modify their internal state (e.g. inventory).

**Comment by byorgey** [comment]
I like this idea.  In https://github.com/swarm-game/swarm/issues/1535#issuecomment-1722511017 I mentioned #462 and #506, but thinking about it more this is actually orthogonal.  Those issues are about keeping track of robots you *know about*; this is about other robot's disposition/receptiveness.  They are orthogonal, since with the two axes known/unknown and friend/foe it is easy to imagine things that would fall into each of the four quadrants.

I like the idea of robots being able to change their disposition towards other robots.  When we make further progress in the open-world game, I can easily imagine other actors you encounter and interact with, and their disposition toward you could change depending on your choices.
